File: app/src/main/java/com/doubean/ford/util/TopItemNoBackgroundUtil.java
package com.doubean.ford.util;

import android.content.Context;

import androidx.annotation.ColorInt;
import androidx.core.graphics.ColorUtils;

import com.doubean.ford.R;

public class TopItemNoBackgroundUtil {
    public static @ColorInt
    int getNoBackground(Context context, int curr, int total) {
        @ColorInt int topItemColor = context.getColor(R.color.top_item);
        @ColorInt int yellow = context.getColor(R.color.douban_yellow);
        return ColorUtils.blendARGB(topItemColor, yellow, curr / (total - 1F));
    }
}

File: app/src/main/java/com/doubean/ford/util/TopItemNoBackgroundUtil.kt
package com.doubean.ford.util

import android.content.Context
import androidx.annotation.ColorInt
import androidx.core.graphics.ColorUtils
import com.doubean.ford.R

object TopItemNoBackgroundUtil {
    @ColorInt
    fun getNoBackground(context: Context, curr: Int, total: Int): Int {
        @ColorInt val topItemColor = context.getColor(R.color.top_item)
        @ColorInt val yellow = context.getColor(R.color.douban_yellow)
        return ColorUtils.blendARGB(topItemColor, yellow, curr / (total - 1f))
    }
}